### PR TITLE
Adding custom mapper for json processing exception

### PR DESCRIPTION
**Pull request by himank**
Custom exception mapper to return bad request instead of returning 500.

### CODE REVIEW

File: server/src/main/java/io/druid/server/initialization/jetty/CustomExceptionMapper.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+@Provider
+public class CustomExceptionMapper implements ExceptionMapper<JsonProcessingException>
+{
+  @Override
+  public Response toResponse(JsonProcessingException arg0)
+  {
+    return Response.status(Response.Status.BAD_REQUEST).entity(ImmutableMap.of("Error", arg0.getMessage())).build();
+  }
+}
```
Comment (xvrl) [inline server/src/main/java/io/druid/server/initialization/jetty/CustomExceptionMapper.java]: missing newline

File: server/src/main/java/io/druid/server/initialization/jetty/CustomExceptionMapper.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+@Provider
+public class CustomExceptionMapper implements ExceptionMapper<JsonProcessingException>
+{
+  @Override
+  public Response toResponse(JsonProcessingException arg0)
```
Comment (xvrl) [inline server/src/main/java/io/druid/server/initialization/jetty/CustomExceptionMapper.java]: is it possible to have a JsonProcessingException raised that would be a legitimate server error?

File: server/src/main/java/io/druid/server/initialization/jetty/CustomExceptionMapper.java (hunk 1 above)
Comment (himank) [inline server/src/main/java/io/druid/server/initialization/jetty/CustomExceptionMapper.java]: I'll add it thanks

File: server/src/main/java/io/druid/server/initialization/jetty/CustomExceptionMapper.java (hunk 2 above)
Comment (himank) [inline server/src/main/java/io/druid/server/initialization/jetty/CustomExceptionMapper.java]: As per the javadoc "Intermediate base class for all problems encountered when processing (parsing, generating) JSON content" but let me rethink over it.

**Comment by himank** [comment]
Updated the pull request with customizing JsonMappingException instead of JsonProcessingException as the former one is more appropriate for problems related to mapping content. Before/After output is below.

Before:
HTTP/1.1 500 Server Error
Date: Thu, 14 Jan 2016 20:26:03 GMT
Cache-Control: must-revalidate,no-cache,no-store
Content-Type: text/html; charset=ISO-8859-1
Content-Length: 550
Connection: close
Server: Jetty(9.2.5.v20141112)

<html>
<head>
<meta http-equiv="Content-Type" content="text/html;charset=ISO-8859-1"/>
<title>Error 500 </title>
</head>
<body>
<h2>HTTP ERROR: 500</h2>
<p>Problem accessing /flurry/v1/retention. Reason:
<pre>    javax.servlet.ServletException: com.fasterxml.jackson.databind.JsonMappingException: Could not resolve type id 'timeseries1' into a subtype of …...
 at [Source: HttpInputOverHTTP@1a5e7a1; line: 1, column: 6]</pre></p>
<hr /><i><small>Powered by Jetty://</small></i>
</body>
</html>

After:
HTTP/1.1 400 Bad Request
Date: Thu, 14 Jan 2016 19:48:39 GMT
Content-Type: application/json
Transfer-Encoding: chunked
Server: Jetty(9.2.5.v20141112)

{"Error":"Could not resolve type id ‘timeseries1' into a subtype of ….. at [Source: HttpInputOverHTTP@2b9d115b; line: 1, column: 6]"}

**Comment by fjy** [comment]
Thanks for the contrib @himank. Will try to review shortly

**Comment by himank** [comment]
@fjy Thanks :)

**Comment by himank** [comment]
@fjy If you get chance could you please review it, Thanks in advance.

File: server/src/main/java/io/druid/server/initialization/jetty/CustomExceptionMapper.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import javax.ws.rs.ext.Provider;
+
+@Provider
+public class CustomExceptionMapper implements ExceptionMapper<JsonMappingException>
+{
+  @Override
+  public Response toResponse(JsonMappingException exception)
+  {
+    return Response.status(Response.Status.BAD_REQUEST)
+                   .entity(ImmutableMap.of("Error", exception.getMessage()))
```
Comment (drcrallen) [inline server/src/main/java/io/druid/server/initialization/jetty/CustomExceptionMapper.java]: other places use `"error"` lower case

Comment (drcrallen) [inline server/src/main/java/io/druid/server/initialization/jetty/CustomExceptionMapper.java]: also should probably be `exception.getMessage() == null ? "unown json mapping exception" : exception.getMessage()` or similar

**Comment by drcrallen** [comment]
:+1: after minor comments

**Comment by drcrallen** [comment]
Thanks for the contrib @himank . Could you please fill out the CLA at http://druid.io/community/cla.html ?

**Comment by himank** [comment]
@drcrallen Thanks for the review, updated as per your comments. Please review when you get chance :)

**Comment by fjy** [comment]
:+1:

**Comment by fjy** [comment]
@himank do you mind following the steps here? https://groups.google.com/forum/#!topic/druid-development/1Zs3Vd9LHCQ so your contribution will be counted :)

**Pull request merged**